src/Watcher.hs

```diff
@@ -1,23 +1,24 @@
-module Watcher where
+module Watcher (watch) where
 
 import qualified Builder
 import CliArguments (Args(..))
-import Config
+import Config (Config(..))
 import Data.Semigroup ((<>))
 import qualified Data.Text as T
 import qualified Data.Text.IO as TIO
 import qualified Notify
-import Notify (Config(..))
 import System.FilePath ()
+import Text.Regex (mkRegex)
 
-watch :: Config.Config -> Args -> IO ()
-watch config args = do
+watch :: Config -> Args -> IO ()
+watch config@Config {source_directory, watch_file_extensions, watch_file_ignore_patterns } args = do
   putStrLn "Watching. Enter '?' to see the help."
   state <-
     Notify.watch
       Notify.Config
-      { pathToWatch = Config.source_directory config
-      , relevantExtensions = [".elm", ".coffee", ".js", ".json"]
+      { pathToWatch = source_directory
+      , relevantExtensions = watch_file_extensions
+      , ignorePatterns = map mkRegex $ map T.unpack watch_file_ignore_patterns
```

Comment:
is there a `mkRegex` that returns an `Either`. would be nice to get a nice error message if the config contains broken regex. and maybe move mkRegex to Config.hs

Reply:
> is there a `mkRegex` that returns an `Either`

Bizarrely enough, there doesn't seem to be one. At least I can't see one in [the docs](https://hackage.haskell.org/package/regex-compat-0.95.1/docs/Text-Regex.html).

> and maybe move mkRegex to Config.hs

I put it in here to avoid the work of compiling the regular expressions – though this is not likely to be much work – until the user requests `--watch` mode. I should have mentioned that in the PR. Do you want me to move it?

Reply:
I think it's okay to keep it this way.